Upgrade to Java 8 - announce the plan

batmat opened the pull request:
Now the CI build does run on Windows too, and is back to green, it's time to act on the plan I think.

@olivergondza @kohsuke @daniel-beck @oleg-nenashev @rtyler @slide @orrc @jtnord @jglick

rtyler commented on content/blog/2017/2017-01-17-Jenkins-is-upgrading-to-Java-8.adoc:
```diff
@@ -0,0 +1,40 @@
+
+See the following for more details about that.
+
+== Timeline
```
I think the timeline should be presented before the previous "Why upgrade to Java 8", as it's more important to the vast majority of users

bitwiseman commented on content/blog/2017/2017-01-17-Jenkins-is-upgrading-to-Java-8.adoc:
```diff
@@ -0,0 +1,40 @@
+Here is how we plan to roll that baseline upgrade in the next few months.
+
+1. Now: Announce the intention publicly.
+2. April, 2017: Drop support for Jenkins weekly.
```
April 2017: Drop support for Java 7 in Jenkins weekly builds.

bitwiseman commented on content/blog/2017/2017-01-17-Jenkins-is-upgrading-to-Java-8.adoc:
```diff
@@ -0,0 +1,40 @@
+1. Now: Announce the intention publicly.
+2. April, 2017: Drop support for Jenkins weekly.
+** We will declare what LTS will be the last one to support Java 7.
+3. April-June: First LTS version requiring Java 8 is published.
```
May 2017:  Publish first LTS version requiring Java 8.

Pick a month.   You can be wrong, but pick a month.

bitwiseman commented on content/blog/2017/2017-01-17-Jenkins-is-upgrading-to-Java-8.adoc:
```diff
@@ -0,0 +1,40 @@
+** We will declare what LTS will be the last one to support Java 7.
+3. April-June: First LTS version requiring Java 8 is published.
+
+CAUTION: So, this means that if you are still running Java 7, you will not be able to upgrade to the latest LTS version after some date probably around May 2017.
```
`s/So, this means that //`
In side a caution block, you don't need the lead in.  

`s/some date probably around/approximately/`

bitwiseman commented on content/blog/2017/2017-01-17-Jenkins-is-upgrading-to-Java-8.adoc:
```diff
@@ -0,0 +1,40 @@
+* Java 7 has been now end-of-lifed for 18+ months ;
+* People are already moving away from Java 7, as show the numbers ;
+* Java 8 runtime is known from the field to be more stable
+** Many developers have been wanting to be allowed to leverage the improvements that Java 8 provides to the language and platform
```
`s/**/*/`
When doing bullets don't make lone bullet.

`s/wanting to be allowed to leverage/wanting to leverage/`

bitwiseman commented on content/blog/2017/2017-01-17-Jenkins-is-upgrading-to-Java-8.adoc:
```diff
@@ -0,0 +1,40 @@
+
+Balancing those numbers with many other criteria:
+
+* Java 7 has been now end-of-lifed for 18+ months ;
```
`s/;//` 
Don't need semicolons in bulleted list.

batmat commented:
@oleg-nenashev @rtyler @bitwiseman addressed your comments.

I calculated the first LTS will be in June finally.

See https://docs.google.com/spreadsheets/d/1CmALlETKlBJC56JmicSUQyX9AFTHFhPPa2c7GV5v4-M/edit#gid=0 for yourself if I didn't miss a line. 

Cheers

bitwiseman commented on content/blog/2017/2017-01-17-Jenkins-is-upgrading-to-Java-8.adoc:
```diff
@@ -0,0 +1,41 @@
+
+**In the next few months, Jenkins will require Java 8 as its runtime.**
+
+Back in link:/blog/2016/11/03/2016-11-22-what-jvm-versions-are-running-jenkins-the-return/[last November, we discussed interesting statistics showing that Jenkins was now running Java 8 on a majority of its running instances].
```
This paragraph is logically connected to https://github.com/jenkins-infra/jenkins.io/pull/545/files#diff-5585914dc691986da9b7537331fe0c1aR31 .  It's more reasons why we're upgrading. 

Can you move these two lines to just before line 31?

Or move this information into the list:

```
* Over 55% of all Jenkins instances are already running on Java 8.
* Over 70% of all Jenkins 2.x instances are running on Java 8.
```

batmat replied:
Done

daniel-beck commented:
It should mention that we don't provide security updates for older release lines, so it's possible (although unlikely) that the first LTS release requiring JRE 8 will also be a security update.

oleg-nenashev commented:
> (although unlikely) that the first LTS release requiring JRE 8 will also be a security update.

I would strongly vote against it. If it is really required, it would make much more sense to spin security releases for two baselines as an exception

daniel-beck commented:
> If it is really required, it would make much more sense to spin security releases for two baselines as an exception

We didn't do that for the 1.625.x line, and I'm against doing that now. (*Note to casual readers: There's currently no security update planned for 2.60.x. It's still too far off to know.*) However I'm in favor of informing users about how we'd be handling this, to give them all relevant information.

rtyler commented:
@batmat what do you think, should this be published today (17th)?

batmat commented:
@rtyler I can adjust things and, well, I don't feel strongly for that very day, but it should be published asap IMO indeed for reasons

batmat commented:
@rtyler from my point of view it's ready. I just pushed a small adjustment to address last @bitwiseman comment.

So if you agree on the content, yes I think it would be great if you can merge.

Thanks!

jtnord commented:
> Now the CI build does run on Windows too, and is back to green

Actually there are failures and the status lies :(
